src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -11,6 +12,10 @@ export default class GeGiftEntryFormApp extends LightningElement {
     @track errorTitle;
     @track errorMessage;
 
+    async connectedCallback() {
+        createDefaultTemplate();
+    }
```

Comment:
Along the lines of @ptewson-sfdo's comment about timing for calling this method, instead of calling it imperatively every time the component is loaded, I wonder if it would make sense to put this inside the `GE_FormRendererService.getDefaultSGERenderWrapper` method.  If no default template is returned from `formService.retrieveDefaultSGEFormTemplate()`, then the `GE_DefaultTemplate.createDefaultTemplate` method could be called there, which would mean a default template would always be returned from `GE_FormRendererService.getDefaultSGERenderWrapper`, and there wouldn't be any additional Apex calls.

Reply:
I keep waffling about this, but I THINK this actually makes sense. The only other thought is that we might want a second catch for it - if you are loading your template list and get 0 results? Otherwise I think we might miss a path where you are only using Batch mode and may not have any templates at all

Reply:
I think adding an extra call at `retrieveDefaultSGEFormTemplate` makes sense, but I don't think that would replace having the call here. I don't think batch situations are covered as mentioned by Patrick. Maybe it makes sense to do the whole create default template process at the beginning of 'retrieveDefaultSGEFormTemplate' so that there will always be a default template, in addition to keeping this call here?